### PR TITLE
Added env variable for cloud-provider

### PR DESCRIPTION
**Pull request by jdnurme**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

This allows the cloud provider to be specified as an environment variable in preparation for cloud provider extraction.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
NONE

```release-note
NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
I don't know the history here, can you please take a look whether `KUBERNETES_PROVIDER` does the same and maybe needs to be aligned. Also I noticed this env var already used in some places. Just want to make sure there are not conflicts or such. lgtm otherwise

**Comment by cheftako** [comment]
> I don't know the history here, can you please take a look whether `KUBERNETES_PROVIDER` does the same and maybe needs to be aligned. Also I noticed this env var already used in some places. Just want to make sure there are not conflicts or such. lgtm otherwise

I believe @jdnurme is trying to have a flag which only controls the cloud-provider flag passed in to Kubelet/KCM/KAS so that we can test the relevant feature gate. (https://github.com/kubernetes/kubernetes/blob/ad99a592f7399ba8b5a1676e11cbed4598806716/pkg/features/kube_features.go#L540)

**Comment by jdnurme** [comment]
> > I don't know the history here, can you please take a look whether `KUBERNETES_PROVIDER` does the same and maybe needs to be aligned. Also I noticed this env var already used in some places. Just want to make sure there are not conflicts or such. lgtm otherwise
> 
> I believe @jdnurme is trying to have a flag which only controls the cloud-provider flag passed in to Kubelet/KCM/KAS so that we can test the relevant feature gate. (
> 
> https://github.com/kubernetes/kubernetes/blob/ad99a592f7399ba8b5a1676e11cbed4598806716/pkg/features/kube_features.go#L540
> 
> )

Correct, this flag is intended for testing with cloud providers disabled.

**Comment by SergeyKanzhelev** [comment]
yes, I understand this. I see that some scripts are using `KUBERNETES_PROVIDER` for the same purpose, thus the question on whether any of them cross-called. I didn't look seep.

Also I see this: https://github.com/kubernetes/kubernetes/blob/ff696d894340e1a4a54a63f26cfbc72c26239f7c/cluster/log-dump/log-dump.sh#L537 and wonder if there may be any confusions with the name of local var and env var.

Again, I didn't see any red flags, just want to make sure we reuse things or clean up if there is a chance.

**Comment by jdnurme** [comment]
> yes, I understand this. I see that some scripts are using `KUBERNETES_PROVIDER` for the same purpose, thus the question on whether any of them cross-called. I didn't look seep.
> 
> Also I see this:
> 
> https://github.com/kubernetes/kubernetes/blob/ff696d894340e1a4a54a63f26cfbc72c26239f7c/cluster/log-dump/log-dump.sh#L537
> 
> and wonder if there may be any confusions with the name of local var and env var.
> Again, I didn't see any red flags, just want to make sure we reuse things or clean up if there is a chance.

I believe that is slightly different, simply because the KUBERNETES_PROVIDER variable seems to be specific to kubernetes platforms. In that link i see "gke" as the value being replaced. Cloud provider is non-specific to whether or not the platform is integrated with kubernetes. In this instance it's replacing "gce". I'm not confident these can be used interchangeably so I think the safest option is to keep "CLOUD_PROVIDER" as separate. In my tests, I am running against GCE, not GKE.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -567,3 +567,7 @@ export WINDOWS_ENABLE_HYPERV="${WINDOWS_ENABLE_HYPERV:-false}"
 # TLS_CIPHER_SUITES defines cipher suites allowed to be used by kube-apiserver.
 # If this variable is unset or empty, kube-apiserver will allow its default set of cipher suites.
 export TLS_CIPHER_SUITES=""
+
+# CLOUD_PROVIDER defines the cloud-provider value presented to KCM, apiserver,
+# and kubelet
+export CLOUD_PROVIDER="${CLOUD_PROVIDER:-gce}"
```
Comment (cheftako) [inline cluster/gce/config-default.sh:573]: The CLOUD_PROVIDER variable is already used [here](https://github.com/kubernetes/kubernetes/blob/master/cluster/kubemark/util.sh#L19). I would suggest switching this to something like CLOUD_PROVIDER_FLAG. Its more descriptive and not yet used.

**Comment by dims** [comment]
@jdnurme please see one pending comment from @cheftako https://github.com/kubernetes/kubernetes/pull/106241#discussion_r772588680

**Pull request merged**